### PR TITLE
Lift state from SongList to redux store

### PR DESCRIPTION
**Pull request by jhb188**
- Also rewrite `changeVote` as redux action to clean up `connect`

### CODE REVIEW

File: src/store/newSong/reducer.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {
+  TOGGLE_IS_EDITING,
+} from './constants'
+
+const initialState = {
+  data: {},
+  isEditing: false,
+}
```
Comment (jhb188) [inline src/store/newSong/reducer.js]: we should use `data` for `NewSongForm` fields

File: src/store/songs/actions.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { CHANGE_VOTE_FAIL } from './constants'
+
+export const changeVote = song => (dispatch, getState, getFirebase) =>
+  async value => {
+    const firebase = getFirebase()
+    const votes = song.votes || 0
+    const newVotes = votes + value
+    try {
+      firebase.update(`/songs/${song.id}`, { votes: newVotes })
```
Comment (jhb188) [inline src/store/songs/actions.js]: this `try` block may be a reasonable place to put `localStorage` vote tracking logic

File: src/store/songs/actions.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { CHANGE_VOTE_FAIL } from './constants'
+
+export const changeVote = song => (dispatch, getState, getFirebase) =>
```
Comment (rossvz) [inline src/store/songs/actions.js]: So this basically is called with a specific song, and returns a function with how to changeVotes for that specific song? Is there a reason we dont dispatch anything on the success case?

Comment (jhb188) [reply src/store/songs/actions.js]: yes, we're using a redux middleware that is automatically dispatching actions and managing updates to the `firebase` slice of redux state in response to firebase data changing. if we were managing that part of state ourselves, we would define our own actions to describe those state updates and dispatch them.

**Pull request merged**